python/pyarrow/tests/test_pandas.py

```diff
@@ -4312,10 +4314,13 @@ def test_array_to_pandas():
 def test_roundtrip_empty_table_with_extension_dtype_index():
     df = pd.DataFrame(index=pd.interval_range(start=0, end=3))
     table = pa.table(df)
-    table.to_pandas().index == pd.Index([{'left': 0, 'right': 1},
-                                         {'left': 1, 'right': 2},
-                                         {'left': 2, 'right': 3}],
-                                        dtype='object')
+    if Version(pd.__version__) > Version("1.0"):
+        tm.assert_index_equal(table.to_pandas().index, df.index)
+    else:
+        assert table.to_pandas().index == pd.Index([{'left': 0, 'right': 1},
```

Comment:
I don't think we still support older pandas versions? (https://github.com/apache/arrow/pull/14631) So then this can be simplified

Reply:
It was failing on pandas 1.0 🤷‍♀️ 
https://github.com/apache/arrow/actions/runs/7530944983/job/20498420418#step:6:6647

Reply:
Ah, sorry I missed the > (_strictly_ greater), not >=, ignore me then ;)